src/api-service/__app__/onefuzzlib/endpoint_authorization.py

```diff
@@ -43,6 +44,35 @@ def is_agent(token_data: UserInfo) -> bool:
     return False
 
 
+def can_modify_config(req: func.HttpRequest, config: InstanceConfig) -> bool:
+    user_info = parse_jwt_token(req)
+    if not isinstance(user_info, UserInfo):
+        return False
+
+    if config.admins is None:
+        return True
+
+    return user_info.object_id in config.admins
+
+
+def check_can_manage_pools(req: func.HttpRequest) -> Optional[Error]:
```

Comment:
I think we need to unit test this and `can_modify()`.

Reply:
@bmc-msft, to be more specific: I suggest some tests that check the relationship between the `InstanceConfig` and `allow_pool_management`.

One way we could do this is have these top-level functions extract the user from the `HttpRequeset`, and then pass it to instance methods on `InstanceConfig`. These could be predicates that return `bool` based on whether or not a user is an admin generally, and if `allow_pool_management` is set, and should be easy to test in isolation.